### PR TITLE
updating status verification

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

**Review by monaullah — CHANGES_REQUESTED** [review]
See description in Jira task:

> then the approval status is set back to Pending and the rejection note that followed the rejection is removed

I think this task is blocked by https://unit.atlassian.net/browse/NP-45218 / https://github.com/BIBSYSDEV/nva-nvi/pull/92

Finish https://github.com/BIBSYSDEV/nva-nvi/pull/92 first?

**Review by ketilaa — APPROVED** [review]

File: nvi-rest/src/test/java/no/sikt/nva/nvi/rest/upsert/UpdateNviCandidateStatusHandlerDbTest.java
Comment (brinxmat) [inline nvi-rest/src/test/java/no/sikt/nva/nvi/rest/upsert/UpdateNviCandidateStatusHandlerDbTest.java:102]: In another PR, we need to be able to easily access a given approval in a sensible way…i.e. what is a unique and relevant way of accessing an approval? Probably, `Optional<ApprovalStatus> approvalStatuses.getApprovalForCustomer(URI customer)`

**Review by brinxmat — APPROVED** [review]
See comment

**Pull request closed without merging**
